app/models/organization_campaign.rb

```diff
@@ -30,7 +30,7 @@ def assigned
   end
 
   def matched
-    recipients.joins(:matches).count
+    recipients.matched.count
```

Comment:
What about a counter_cache column?

http://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache